### PR TITLE
Docs/amqp

### PR DESCRIPTION
**Pull request by Sypheos**
Starting amqp doc. soon to be added into the website

issue #356

### CODE REVIEW

**Comment by johanstokking** [comment]
Nice. Make it more aligned with the MQTT docs though in terms of styling and explanation

File: amqp/README.md [hunk 1]
```diff
@@ -1,51 +1,41 @@
...
-========================
+# API Reference
 
-If you don't know what AMQP is you can read [this](https://blogs.vmware.com/vfabric/2013/02/choosing-your-messaging-protocol-amqp-mqtt-or-stomp.html)
-and [this](https://www.rabbitmq.com/tutorials/amqp-concepts.html).
+TTN makes use of an AMQP Topic exchange (`amq.topic`). See [this](https://www.rabbitmq.com/tutorials/amqp-concepts.html) for details.
 
-TTN use the Topic AMQP exchange. The default exchange name is `amq.topic`
+AMQP is not yet available in the public network. The main reasons for this are: 
+_security_, _scalability_ and the AMQP protocol not being _standardised_ yet.
```
Comment (johanstokking) [inline amqp/README.md]: It is, but not the broker that we're using. AMQP 1.0 is a decent, standardised protocol. ActiveMQ is a nice broker for it, but we don't have a Go client for it

File: amqp/README.md [hunk 2]
```diff
@@ -1,51 +1,41 @@
...
-* Match all App devices uplink: `<App_ID>.devices.*.up`
-
-## Uplink
+* Host: The address of the handler on which your application is registered.
+* Port: `5672`
+* Exchange: `ttn.handler`
 
+## Uplink Messages
 
 **Routing key:** `<App_ID>.devices.<Dev_ID>.up`
```
Comment (johanstokking) [inline amqp/README.md]: No underscores; this is the reference https://www.thethingsnetwork.org/docs/applications/mqtt/api.html

File: amqp/README.md [hunk 3]
```diff
@@ -121,11 +139,10 @@ func main() {
 }
 ```
 
-**Usages**
-
-* **rabbitMQ:** `'rabbitmqadmin publish exchange=ttn.handler routing_key=app-id.devices.dev-id.down payload={"port":1,"payload_raw":"AQIDBA=="}'`
+**Usage (RabbitMQ):**
+`rabbitmqadmin publish exchange='ttn.handler' routing_key='app-id.devices.dev-id.down' payload='{"port":1,"payload_raw":"AQIDBA=="}'`
```
Comment (johanstokking) [inline amqp/README.md]: Is it clear that `app-id` and `dev-id` are fields?

File: mqtt/README.md [hunk 4]
```diff
@@ -125,7 +127,8 @@ you will see this on MQTT:
 }
 ```
 
-**Usage (Mosquitto):** `mosquitto_pub -h <Region>.thethings.network:1883 -d -t 'my-app-id/devices/my-dev-id/down' -m '{"port":1,"payload_raw":"AQIDBA=="}'`
+**Usage (Mosquitto):**
+`mosquitto_pub -h <Region>.thethings.network:1883 -d -t 'my-app-id/devices/my-dev-id/down' -m '{"port":1,"payload_raw":"AQIDBA=="}'`
```
Comment (johanstokking) [inline mqtt/README.md]: Avoid unrelated changes in a PR

File: amqp/README.md (hunk 1 above)
Comment (jpmeijers) [inline amqp/README.md]: Should we remove this entire paragraph, or trim it down to "AMQP is not yet available in the public network."

Comment (johanstokking) [inline amqp/README.md]: Yes

File: amqp/README.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
+# API Reference
+
+TTN makes use of an AMQP Topic exchange (`amq.topic`). See [this](https://www.rabbitmq.com/tutorials/amqp-concepts.html) for details.
```
Comment (johanstokking) [inline amqp/README.md]: It is of _type_ topic exchange, but it is not the builtin `amq.topic` exchange but as mentioned below the `ttn.handler` exchange

File: amqp/README.md [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+channel.start_consuming()
+```
+
+
+## Downlink Messages
+
+**Routing key:** `<AppID>.devices.<DevID>.down`
+
+**Message:**
+```json
```
Comment (johanstokking) [inline amqp/README.md]: `js` to get rid of GitHub syntax errors

File: amqp/README.md [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+  }
+	err = s.SubscribeAppEvents("my-app-id", "some-event",
+			func(_ Subscriber, appID string, eventType types.EventType, payload []byte) {
+			  // Do your stuff
+			})
+	//...
+}
+```
+
+*Publish:*
```
Comment (johanstokking) [inline amqp/README.md]: We won't allow applications to publish device events themselves so this example can be removed

File: amqp/README.md [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+    "modulation": "LORA",
+    "data_rate": "SF7BW125",
+    "counter": 123,
+    "frequency": 868300000,
+    "power": 14
+  }
+}
+```
+
+**Usages (GO)**
```
Comment (johanstokking) [inline amqp/README.md]: Usage (Go client)

**Comment by jpmeijers** [comment]
Personally I like Python examples because they are normally a lot quicker to get working than a Go example. I can easily create MQTT examples in Python too. I don't think writing an entire client in Python is necessary, but a few examples I find useful.

Ultimately your choice. Either more Python examples, or I remove all of them.

File: amqp/README.md (hunk 5 above)
Comment (jpmeijers) [inline amqp/README.md]: Ahh ok. I was unsure about this.

**Comment by johanstokking** [comment]
Agreed. However adding code examples on a page where we basically explain the _protocol_, is a sliding edge.

We already cover it in the docs, see:

- Java: https://www.thethingsnetwork.org/docs/applications/java/quick-start.html
- Node.js https://www.thethingsnetwork.org/docs/applications/nodejs/quick-start.html

So then we have Go and Python here, and Java and Node.js elsewhere. I know I'm totally hysterical about consistency, so I'm open to suggestions, but let's leave the protocol here and link to code examples elsewhere
